object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,24 @@
+/* exported pick */
+/*
+input: source = object & keys - an array of the property name
+
+new empty object so that we can set the property keys and values
+
+Object.keys will return an array of the keys of the given object
+loop through the array of keys
+check if the key is found in the source object
+if it is, we want to set the key and value for the matched pair in our new object "pickedObject"
+
+Goal: make a new object with specified property and value
+*/
+
+function pick(source, keys) {
+  var pickedObject = {};
+  var sourceObjectKeys = Object.keys(source); // becomes an array of keys in the source
+  for (var i = 0; i < Object.length + 1; i++) {
```

Comment:
I have **_no idea_** how this works. As this is written, this should only ever run for two iterations, and those iterations have nothing to do with the input object `source`.